openedx/envs/common.py

```diff
@@ -0,0 +1,691 @@
+        'service_user': '800/minute',
+        'registration_validation': '30/minute',
+        'high_service_user': '2000/minute',
+    },
+}
+
+################################ Heartbeat #################################
+
+# Checks run in normal mode by the heartbeat djangoapp
+HEARTBEAT_CHECKS = [
+    'openedx.core.djangoapps.heartbeat.default_checks.check_modulestore',
+    'openedx.core.djangoapps.heartbeat.default_checks.check_database',
+]
+
+# Other checks to run by default in "extended"/heavy mode
+HEARTBEAT_EXTENDED_CHECKS = (
+    'openedx.core.djangoapps.heartbeat.default_checks.check_celery',
+)
+
+HEARTBEAT_CELERY_TIMEOUT = 5
+
+############################ RedirectMiddleware ############################
+
+# Setting this to None causes Redirect data to never expire
+# The cache is cleared when Redirect models are saved/deleted
+REDIRECT_CACHE_TIMEOUT = None  # The length of time we cache Redirect model data
+REDIRECT_CACHE_KEY_PREFIX = 'redirects'
+
+########################### Django Debug Toolbar ###########################
+
+# We don't enable Django Debug Toolbar universally, but whenever we do, we want
+# to avoid patching settings.  Patched settings can cause circular import
+# problems: https://django-debug-toolbar.readthedocs.org/en/1.0/installation.html#explicit-setup
+
+DEBUG_TOOLBAR_PATCH_SETTINGS = False
+
+################ Shared Functions for Derived Configuration ################
+
+
+def make_mako_template_dirs(settings):
+    """
+    Derives the final list of Mako template directories based on the provided settings.
+
+    Args:
+        settings: A Django settings module object.
+
+    Returns:
+        list: A list of Mako template directories, potentially updated with additional
+        theme directories.
+    """
+    if settings.ENABLE_COMPREHENSIVE_THEMING:
+        themes_dirs = get_theme_base_dirs_from_settings(settings.COMPREHENSIVE_THEME_DIRS)
+        for theme in get_themes_unchecked(themes_dirs, settings.PROJECT_ROOT):
+            if theme.themes_base_dir not in settings.MAKO_TEMPLATE_DIRS_BASE:
+                settings.MAKO_TEMPLATE_DIRS_BASE.insert(0, theme.themes_base_dir)
+    return settings.MAKO_TEMPLATE_DIRS_BASE
+
+
+def make_locale_paths(settings):
+    """
+    Constructs a list of paths to locale directories used for translation.
+
+    Localization (l10n) strings (e.g. django.po) are found in these directories.
+
+    Args:
+        settings: A Django settings module object.
+
+    Returns:
+        list: A list of paths, `str` or `path.Path`, to locale directories.
+    """
+    locale_paths = list(settings.PREPEND_LOCALE_PATHS)
+    locale_paths += [settings.REPO_ROOT + '/conf/locale']  # edx-platform/conf/locale/
+
+    if settings.ENABLE_COMPREHENSIVE_THEMING:
+        # Add locale paths to settings for comprehensive theming.
+        for locale_path in settings.COMPREHENSIVE_THEME_LOCALE_PATHS:
+            locale_paths += (path(locale_path), )
+    return locale_paths
```

Comment:
I agree with making make_mako_template_dirs a public function, as both lms/ and cms/ envs need to call it in order to define two different MAKO_TEMPLATE_DIRS (although, I'm not sure that the difference between the lms and cms default for this setting is functionally significant, but we don't have to untangle that for this PR).

For make_locale_paths, though, it looks like both LMS and CMS are using it in the exact same way (`LOCALE_PATHS = Derived(_make_locale_paths)`). In order to keep the public interface as a small as possible, how about keeping this method private, and moving the definition of LOCALE_PATHS up into openedx/envs/common.py?